packages/sdk/src/deserialization.ts

```diff
@@ -88,3 +92,242 @@ export function deserializeAccountTransaction(serializedTransaction: Cursor): {
+    const proofIdCredPubLength = serializedPayload.read(4).readUInt32BE(0);
+    for(let a = 0; a < proofIdCredPubLength; a++) {
+        /*const arIdentity = */serializedPayload.read(4);
+        /*const comEncEqResponse = */serializedPayload.read(96);
+    }
+
+    //  start of IdOwnershipProofs.proofIpSig
+    /*const responseRho = */serializedPayload.read(32);
+    const proofLength = serializedPayload.read(4).readUInt32BE(0);
+    //length x (F, F)
+    for(let a = 0; a < proofLength; a++) {
+        /*const firstF = */serializedPayload.read(32);
+        /*const secondF = */serializedPayload.read(32);
+    }
+    //  end of IdOwnershipProofs.proofIpSig
+
+    //IdOwnershipProofs.proofRegId
+    serializedPayload.read(160);
+
+    //IdOwnershipProofs.proofCredCounter
+    serializedPayload.read(48*4); //4 times 48, g1Elements
+    serializedPayload.read(32*3); //3 times 32, scalars1
+    
+    const groupElementLength = serializedPayload.read(4).readUInt32BE(0);
+    for(let a = 0; a < groupElementLength; a++) {
+        serializedPayload.read(48);
+        serializedPayload.read(48);
+    }
+
+    serializedPayload.read(32*2) //2 times 32, scalars2
+
+    //AccountOwnershipProof
+    const numberOfSignatures = serializedPayload.read(1).readUInt8(0);
+
+    for(let a = 0; a < numberOfSignatures; a++) {
+        //AccountOwnershipProofEntry
+        /*const index = */serializedPayload.read(1);
+        /*const sig = */serializedPayload.read(64);
+    }
+
+    //populate placeholder, if any can be populated, and go back to the for loop in deserialize() and read next CredentialDeploymentInformation, if any    
+    if(data.newCredentials) {
+        //TODO: is cmmCredCounter the same as bluepaper.credCounter inside CredentialDeploymentCommitments
+        data.newCredentials[currentLocation].cdi.commitments.cmmCredCounter = credCounter.toString(); 
+    }
+}
+
+export function deserializeCredentialsToBeRemoved(serializedPayload: Cursor, data: Partial<UpdateCredentialsPayload>) {
+    
+    //TransactionPayload.UpdateCredentials.removeLength
+    //number of credentials to be removed
+    const removeLength = serializedPayload.read(1).readUInt8(0);
+
+    //TransactionPayload.UpdateCredentials.CredIds
+    const removeCredIds: string[] = [];
+    //the credential IDs of the credentials to be removed, based on the removeLength value
+    for(let a = 0; a < removeLength; a++) {
+        const credentialRegistrationId = serializedPayload.read(48);
+        removeCredIds[a] = credentialRegistrationId.toString();
+    }
+
+    //TransactionPayload.UpdateCredentials.newThresholdd
+    //AccountThreshold
+    const newThreshold = serializedPayload.read(1).readUInt8(0);
+
+    //populate placeholder
+    data.removeCredentialIds = removeCredIds;
+    data.threshold = newThreshold;
+
+    //TODO:This is not part of the transaction model, so no value to deserialize, will be passing an empty value for now
+    data.currentNumberOfCredentials = undefined;
```

Comment:
You're right, this should also be removed. I've created something to replace these legacy types in #602, which is basically the notion of transaction "input" types which are added to the `AccountTransactionInput` union type.

I think it'd make sense to get #602 merged and rebase your work here on top of that (or simply merge it in if rebase gets difficult).